docs_src/design/adr/0023-North-South-Messaging.md

```diff
@@ -0,0 +1,527 @@
+
+``` JSON
+{
+    "Correlation-ID": "14a42ea6-c394-41c3-8bcd-a29b9f5e6835",
+    "apiVersion": "v2",
+    "requestId": "e6e8a2f4-eb14-4649-9e2b-175247911369",
+    "queryParams": {
+        "offset":0,
+        "limit":20,
+     }
+    "payload": 
+    {
+    }
+}
+
+In the example query to get commands for a specific device by name, the device name would be in the topic, so the query message would be without information.
+
+{
+    "Correlation-ID": "14a42ea6-c394-41c3-8bcd-a29b9f5e6835",
+    "apiVersion": "v2",
+    "requestId": "e6e8a2f4-eb14-4649-9e2b-175247911369",
+    "queryParams": {
+     }
+    "payload": 
+    {
+    }
+}
+
+```
+
+The **response** message `payload` for queries would contain the information necessary to make a message-based command request.
+
+An example response message is shown below.  Again, note that the message `envelope` wraps the response `payload`.
+
+``` JSON
+{
+    "Correlation-ID": "14a42ea6-c394-41c3-8bcd-a29b9f5e6835",
+    "apiVersion": "v2",
+    "requestId": "e6e8a2f4-eb14-4649-9e2b-175247911369",
+    "errorCode": 0,
+    "payload": 
+    {
+        "apiVersion": "v2",
+        "deviceCoreCommands": [
+            {
+            "deviceName": "testDevice1",
+            "profileName": "testProfile",
+            "coreCommands": [
+                {
+                "name": "coolingpoint1",
+                "get": true,
+                "topic": "/edgex/command/request/testDevice1/coolingpoint1/get",
+                "url": "broker.address:1883",
+                "parameters": [
+                    {
+                    "resourceName": "resource1",
+                    "valueType": "Int32"
+                    }
+                ]
+                }
+            ]
+            },
+            {
+            "deviceName": "testDevice1",
+            "profileName": "testProfile",
+            "coreCommands": [
+                {
+                "name": "coolingpoint1",
+                "set": true,
+                "topic": "/edgex/command/request/testDevice1/coolingpoint1/set",
+                "url": "broker.address:1883",
+                "parameters": [
+                    {
+                    "resourceName": "resource5",
+                    "valueType": "String"
+                    },
+                    {
+                    "resourceName": "resource6",
+                    "valueType": "Bool"
+                    }
+                ]
+                }
+            ]
+            }
+        ]
+    }
+}
+
+```
+
+### Topic Naming
+
+![image](command-service-topics.png)
+
+#### 3rd party system topics
+
+The 3rd party system or application must publish command requests messages to its own MQTT topic (**external message bus**) and subscribe to responses from the same.  Messages topics should follow a standard such as:
```

Comment:
I thought we discussed this and said that its not really up to the 3rd party system to define these topics? It's our mechanism, so we should define the topics the 3rd party applications should use when they want to use Core Command.

Reply:
overlooked text that was not updated. fixed.